packages/ckeditor5-link/src/linkediting.ts

```diff
@@ -280,7 +280,9 @@ export default class LinkEditing extends Plugin {
 			evt.stop();
 			data.preventDefault();
 
-			openLink( url );
+			const bookmarkEditing = editor.plugins.has( 'BookmarkEditing' ) ? editor.plugins.get( 'BookmarkEditing' ) : null;
+
+			handleLinkOpening( url, bookmarkEditing );
```

Comment:
I'd extract it above those listeners and use sort of a factory to create link opener.